interpreter/src/objects.rs

```diff
@@ -535,7 +618,23 @@ impl Value {
                     map: Arc::from(map),
                 }))
             }
-            Expression::Ident(name) => ctx.get_variable(&***name),
+            Expression::Ident(name) => {
+                // First check if this is a variable in the context
+                let var_result = ctx.get_variable(&***name);
+                if var_result.is_ok() {
+                    // If it's a variable, use that value
+                    var_result
+                } else {
+                    // Otherwise, check if it's a type identifier
```

Comment:
`list[10]` with context `list: []` returns

```
failed to create CEL env: overlapping identifier for name 'list'
```

Reply:
Yeah using `list` was a bad example on my part because that's a reserved keyword. My main concern was the removal of the error for attempting to index into a list with a non-int.